cmd/goupnpdcpgen/metadata.go

```diff
@@ -1,69 +1,116 @@
 package main
 
+import (
+	"strings"
+)
+
 // DCP contains extra metadata to use when generating DCP source files.
 type DCPMetadata struct {
 	Name         string // What to name the Go DCP package.
 	OfficialName string // Official name for the DCP.
-	DocURL       string // Optional - URL for further documentation about the DCP.
-	XMLSpecURL   string // Where to download the XML spec from.
-	// Any special-case functions to run against the DCP before writing it out.
-	Hacks []DCPHackFn
+	Src          dcpProvider
 }
 
 var dcpMetadata = []DCPMetadata{
 	{
 		Name:         "internetgateway1",
 		OfficialName: "Internet Gateway Device v1",
-		DocURL:       "http://upnp.org/specs/gw/UPnP-gw-InternetGatewayDevice-v1-Device.pdf",
-		XMLSpecURL:   "http://upnp.org/specs/gw/UPnP-gw-IGD-TestFiles-20010921.zip",
-		Hacks:        []DCPHackFn{totalBytesHack},
+		Src: upnpdotorg{
+			DocURL:     "http://upnp.org/specs/gw/UPnP-gw-InternetGatewayDevice-v1-Device.pdf",
+			XMLSpecURL: "http://upnp.org/specs/gw/UPnP-gw-IGD-TestFiles-20010921.zip",
+			Hacks: []DCPHackFn{
+				totalBytesHack("urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1"),
+			},
+		},
 	},
 	{
 		Name:         "internetgateway2",
 		OfficialName: "Internet Gateway Device v2",
-		DocURL:       "http://upnp.org/specs/gw/UPnP-gw-InternetGatewayDevice-v2-Device.pdf",
-		XMLSpecURL:   "http://upnp.org/specs/gw/UPnP-gw-IGD-Testfiles-20110224.zip",
-		Hacks: []DCPHackFn{
-			func(dcp *DCP) error {
-				missingURN := "urn:schemas-upnp-org:service:WANIPv6FirewallControl:1"
-				if _, ok := dcp.ServiceTypes[missingURN]; ok {
-					return nil
-				}
-				urnParts, err := extractURNParts(missingURN, serviceURNPrefix)
-				if err != nil {
-					return err
-				}
-				dcp.ServiceTypes[missingURN] = urnParts
-				return nil
-			}, totalBytesHack,
+		Src: upnpdotorg{
+			DocURL:     "http://upnp.org/specs/gw/UPnP-gw-InternetGatewayDevice-v2-Device.pdf",
+			XMLSpecURL: "http://upnp.org/specs/gw/UPnP-gw-IGD-Testfiles-20110224.zip",
+			Hacks: []DCPHackFn{
+				fixMissingURN("urn:schemas-upnp-org:service:WANIPv6FirewallControl:1"),
+				totalBytesHack("urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1"),
+			},
 		},
 	},
 	{
 		Name:         "av1",
 		OfficialName: "MediaServer v1 and MediaRenderer v1",
-		DocURL:       "http://upnp.org/specs/av/av1/",
-		XMLSpecURL:   "http://upnp.org/specs/av/UPnP-av-TestFiles-20070927.zip",
+		Src: upnpdotorg{
+			DocURL:     "http://upnp.org/specs/av/av1/",
+			XMLSpecURL: "http://upnp.org/specs/av/UPnP-av-TestFiles-20070927.zip",
+		},
+	},
+	{
+		Name:         "ocf/internetgateway1",
```

Comment:
Come to think of it, I think we can ditch `ocf/internetgateway1`. It's just a subset of `ocf/internetgateway2`.

In hindsight I suspect that we never really needed `dcps/internetgateway1`, but I think we're stuck with it for backwards compatibility.

Reply:
According to the standard, that should do it. 

On the other hand, the compiler is able to strip that code from the output binary when it is unused, thus, overall, this has very little impact.

Reply:
True, but the package seems like noise that users could simply ignore, so why generate it at all?

Reply:
Sure thing, i was just advocating. Though, i could not understand why they keep publishing igw1 if igw2 suffice.

Reply:
this has been removed